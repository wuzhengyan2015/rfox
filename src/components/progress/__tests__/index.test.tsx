import { render, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import Progress from '../index'

configure({ adapter: new Adapter() })

describe('Progress tests', () => { 
    it('renders correctly', () => {
        const tree = render(
            <Progress type="line" percent={80} gapDegree={90} />
        )
        expect(tree).toMatchSnapshot()
    })
})