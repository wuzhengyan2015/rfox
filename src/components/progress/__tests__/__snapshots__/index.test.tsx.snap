// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress tests renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "rfox-progress rfox-progress-line rfox-progress--normal",
      "id": "rfox-progress",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "rfox-progress-outer",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "rfox-progress-inner",
              "style": "height:8px",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "rfox-progress-bg",
                  "style": "width:80%",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "rfox-progress-text",
              },
              "children": Array [
                Object {
                  "data": "80%",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "rfox-progress-text",
            },
            "children": Array [
              Object {
                "data": "80%",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "rfox-progress-inner",
                "style": "height:8px",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "rfox-progress-bg",
                    "style": "width:80%",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
